### DISCUSSION TITLE
Runnerautoscaling Failure - Message - "Failed to list workflow runs" GET https://api.github.com/repos/org-name/repo-name/actions/runs?per_page=100&status=queued: 404 Not Found []"

### DISCUSSION THREAD

**Discussion by sabhika** [actions/actions-runner-controller#1701] [Questions]

**Comment by sabhika** [comment]


**Reply by mumoshu** [reply]
@sabhika You need to follow a standard procedure to use a GitHub App, like installing the app onto your organization so that ARC as a GitHub App can access the API. You may already know but in many cases `404` the error returned not only when it's actually inexsitent but also you provided a wrong credential so the endpoint tried it's best not to reveal the existence of the API resource to the unauthenticated user(you and your ARC in this case). ARC works only when a correct credential is provided. So if you're seeing auth errors like that, you need to figure out why..

RunnerDeployment manages Runners so you'd need to check your Runners resources generated by ARC from your RunnerDeployment. See what's in `Runner.Spec.Labels`, not `Runner.Metadata.Labels`. The latter is for K8s pod labels.

**Reply by sabhika** [reply]
@mumoshu  I am still seeing this issue. I have tried to do the sample deployment with the default image and everything works fine. But when I use custom image, I don't see the label being added, Although I see the label in the runner file. 

at runner.spec.labels

So, for testing purpose I tried to add the image in the default runner file and save it, I could see the label was added. 

Hence, I came to conclusion that with my custom image runner deployment I am not able to add the label, due to which I am not able to run the workflows.

Can you please assist?

Runner version: 0.25.2
Kubernetes version: 1.22
cert-manager: 1.91

**Reply by sabhika** [reply]
This is my runner file


apiVersion: actions.summerwind.dev/v1alpha1
kind: Runner
metadata:
  annotations:
    sync-time: "2022-08-31T18:50:34Z"
  creationTimestamp: "2022-08-31T18:50:34Z"
  finalizers:
  - runner.actions.summerwind.dev
  generateName: java-runner-xn7fx-
  generation: 1
  labels:
    runner-deployment-name: java-runner
    runner-template-hash: 67cf6b8f54
  name: java-runner-xn7fx-2prw4
  namespace: actions
  ownerReferences:
  - apiVersion: actions.summerwind.dev/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: RunnerReplicaSet
    name: java-runner-xn7fx
    uid: 13f5ecdb-2ee8-412d-a478-330913784b3e
  resourceVersion: "2324207"
  uid: 5db3c494-a340-4d28-9035-72ac88554a43
spec:
  dockerdContainerResources: {}
  image: my-custom-image-here
  labels:
  - custom-runner
  organization: my-org-here
  resources: {}
status:
  phase: Running
  ready: true
  registration:
    expiresAt: "2022-08-31T19:41:16Z"
    labels:
    - custom-runner
    organization: my-org-here
    token: my-token-here

**Reply by sabhika** [reply]
@jnewland @acj  @damacus Anyone can please look into this?

**Reply by toast-gear** [reply]
> But when I use custom image, I don't see the label being added, Although I see the label in the runner file.

You should probably provide your custom image Dockerfile and highlight any other changes you've made

**Reply by toast-gear** [reply]
why are you using a custom entrypoint? I don't understand what value that is bringing, what does the existing entrypoint not do meaning you'd prefer to use a custom one?

EDIT to expand, ARC isn't a general purpose GitHub Actions runner orchestrator where you can throw any image at it. You either need to build on-top of our images (https://github.com/actions-runner-controller/actions-runner-controller#software-installed-in-the-runner-image) or use our images directly. You could do an entirely custom image and not build on ours entirely but you would still probably want to use our entrypoint + startup scripts or at the very least replicate what our entrypoint is doing in terms of the actions/runner bootstrap set-up.

**Reply by sabhika** [reply]
After changing the base image to "summerwind/actions-runner:latest" my issue got resolved. Thanks for your support. You may close this.

**Reply by toast-gear** [reply]
You are able to build your own image, just don't define a entrypoint in that custom image https://github.com/actions-runner-controller/actions-runner-controller#software-installed-in-the-runner-image

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
